libsolidity/codegen/CompilerContext.cpp

```diff
@@ -356,22 +356,29 @@ CompilerContext& CompilerContext::appendConditionalPanic(util::PanicCode _code)
 
 CompilerContext& CompilerContext::appendRevert(string const& _message)
 {
-	appendInlineAssembly("{ " + revertReasonIfDebug(_message) + " }");
+	callYulFunction(m_yulUtilFunctions.revertReasonIfDebug(_message), 0, 0);
 	return *this;
 }
 
 CompilerContext& CompilerContext::appendConditionalRevert(bool _forwardReturnData, string const& _message)
 {
 	if (_forwardReturnData && m_evmVersion.supportsReturndata())
+	{
+		solAssert(_message.empty(), "");
 		appendInlineAssembly(R"({
 			if condition {
 				returndatacopy(0, 0, returndatasize())
```

Comment:
This violates the memory model, but it's old codegen.

Reply:
And it's only optimized on its own and not in combination with other yul code, right? And we don't plan on doing stack limit evasion in those anyways?
Also we did have the plan of potentially running stack limit evasion on old codegen + abiv2 code at some point - but that's still fine, as long as the abiv2 functions don't call this here, (which they don't)?

Reply:
Yes.